### ISSUE TITLE
story(sbb-carousel): initial implementation

### ISSUE THREAD

**Issue by kyubisation** [sbb-design-systems/lyne-components#3872]
We want to implement a carousel component.

Figma: https://www.figma.com/design/mWknI2rC5DJmOgRO61WKai/Lyne-Components?m=auto&node-id=7491-19
Accessibility Pattern: https://www.w3.org/WAI/ARIA/apg/patterns/carousel/

**Design Specification**

- Use scroll snap CSS feature: https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_scroll_snap
- The carousel does not have a fixed size. It should adopt the width of the first child element. The height is implicitly given by the slotted content.
  - Making sure the content/images align in size is the responsibility of the consumer. This should be clearly documented.
- The carousel should not have a visible scroll bar, but it should support (horizontal) scrolling.

**Technical Specification**

- An event `beforeshow` should be dispatched on the `<sbb-carousel-item>` element when it starts becoming visible and an event `show` when scrolling is finished (ideally after snap). This should be done via IntersectionObserver (and maybe `scrollend` event).
- Make the `<sbb-carousel-list>` host the scroll context.
- The paginator should be automatically configured (accessibility, amount of entries and index)
- Except for the paginator, no keyboard interaction needs to be implemented

Example

```html
<sbb-carousel>
  <sbb-carousel-list>
    <sbb-carousel-item>
      <img src="...">
    </sbb-carousel-item>
    <sbb-carousel-item>
      <sbb-image src="..."></sbb-image>
    </sbb-carousel-item>
    <sbb-carousel-item>
      <picture>
        <source
          srcset="/shared-assets/images/examples/surfer.jpg"
          media="(orientation: portrait)" />
          <img src="/shared-assets/images/examples/painted-hand.jpg" alt="" />
      </picture>
    </sbb-carousel-item>
    <sbb-carousel-item>
      <div>
        Some special content with text
        <img src="...">
      </div>
    </sbb-carousel-item>
  </sbb-carousel-list>
  <sbb-compact-paginator></sbb-compact-paginator>
</sbb-carousel>
```

**Definition of Done**

- [ ] Component is implemented
- [ ] Documentation is provided
- [ ] Tests are implemented
- [ ] Storybook stories are implemented
- [ ] Navigation via keyboard is tested
- [ ] Screen reader output is tested
- [ ] High-contrast is tested
- [ ] Remaining accessibility is tested
- [ ] UX approved
